mesonbuild/utils/universal.py

```diff
@@ -1893,17 +1893,20 @@ class ProgressBarFallback:  # lgtm [py/iter-returns-non-self]
     __iter__ method' warning.
     '''
     def __init__(self, iterable: T.Optional[T.Iterable[str]] = None, total: T.Optional[int] = None,
-                 bar_type: T.Optional[str] = None, desc: T.Optional[str] = None):
+                 bar_type: T.Optional[str] = None, desc: T.Optional[str] = None,
+                 disable: T.Optional[bool] = None):
```

Comment:
It might be easier to reason about what the various values of disable meant if it was an enum. I am not a huge fan of optional booleans. I think it really hurts readability of code.

Reply:
That class mimicks `tqdm` API:

> disable: bool, optional
Whether to disable the entire progressbar wrapper [default: False]. If set to None, disable on non-TTY. 

https://tqdm.github.io/docs/tqdm/

Reply:
nevermind then